rero_ils/modules/entities/replace.py

```diff
@@ -0,0 +1,236 @@
+                                    Entity.create(
+                                        data=new_mef_data,
+                                        dbcommit=True,
+                                        reindex=True
+                                    )
+                                self.logger.info(
+                                    f'Create a new MEF {mef_type} '
+                                    f'record(pid: {mef_data["pid"]})'
+                                )
+                            entity_authorized_access_point = entity[
+                                "entity"]["authorized_access_point"]
+                            mef_authorized_access_point = mef_data[
+                                new_source]["authorized_access_point"]
+                            self.logger.info(
+                                f'Replace document:{doc_pid} '
+                                f'{self.entity_type} '
+                                f'"{entity_authorized_access_point}" - '
+                                f'({mef_type}) {new_source}:{new_source_pid} '
+                                f'"{mef_authorized_access_point}"'
+                            )
+                            entity['entity'] = {
+                                '$ref': (
+                                    f'{self._get_base_url(mef_type)}'
+                                    f'/{new_source}/{new_source_pid}'
+                                ),
+                                'pid': mef_data['pid']
+                            }
+                            changed = True
+                        else:
+                            authorized_access_point = mef_data.get(
+                                source, {}).get('authorized_access_point')
+                            info = f'{doc_entity_type}: {authorized_access_point}'
+                            self.rero_only[identifier] = info
+                            self.logger.warning(
+                                f'No other source found for document:{doc_pid} '
+                                f'{self.entity_type} - '
+                                f'({mef_type}) {identifier} "{info}"'
+                            )
+                    else:
+                        authorized_access_point = entity[
+                            'entity']['authorized_access_point']
+                        info = f'{doc_entity_type}: {authorized_access_point}'
+                        self.not_found[identifier] = info
+                        self.logger.warning(
+                            f'No MEF found for document:{doc_pid} '
+                            f'{self.entity_type} - '
+                            f'({mef_type}) {identifier} "{info}"'
+                        )
+            except Exception as err:
+                self.logger.error(
+                    f'Error document:{doc_pid} {entity} {err}"'
+                )
+        if changed:
+            return doc
+
+    def replace(self):
```

Comment:
Personally, I prefer `def run(self)` for this kind of method (seems more generic)